lm_eval/tasks/ja/xlsum_ja.py

```diff
@@ -89,19 +90,22 @@ def preprocess_ctx(self, ctx, max_length, ctx_prompt="ニュース記事:", summ
         ctxs = [f"{ctx_prompt}{c}"for c in ctx.split(ctx_prompt)]
         description = ""
         if summary_prompt not in ctxs[0]:
-            description = ctxs[0]
+            description = ctxs[0].replace(ctx_prompt, "")
             ctxs = ctxs[1:]
         max_length_per_shot = max_length // len(ctxs)
         res = description
         for c in ctxs:
             text, summary = c.split(summary_prompt)
             sentences = text.split("。")
             c_res = ""
+            add_sentences = []
             for s in sentences:
-                if len(self._tokenize(text=c_res+s)) > max_length_per_shot:
-                    c_res += "\n"
+                tmp = add_sentences + [s]
+                if len(self._tokenize(text="。".join(tmp))) > max_length_per_shot:
+                    add_sentences[-1] += "。"+self.SEP
```

Comment:
non-blocking nit: is there any risk that `if len(self._tokenize(text="。".join(tmp))) > max_length_per_shot:` on L104 evaluates to true when `add_sentences` is an empty list, and `add_sentences[-1]` will throw a `IndexError`?

Reply:
@mrorii thank you for your comment. I believe it's not a case for general tokenizers. But, I made sure to avoid this error  https://github.com/Stability-AI/lm-evaluation-harness/pull/64/commits/e2f2770d6318342ef3229a076db8f2693ddd2789